File: HarmonyOSExample/entry/src/main/java/com/minkiapps/hos/migration/di/modules/ApiModule.java
package com.minkiapps.hos.migration.di.modules;

import com.minkiapps.shared.api.ApiService;
import com.minkiapps.shared.api.FactoryKt;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class ApiModule {

    @Provides
    @Singleton
    public static ApiService providesApiService() {
        return FactoryKt.createApiService();
    }
}

File: AndroidExample/app/src/main/java/com/minkiapps/android/example/di/modules/ApiModule.kt
package com.minkiapps.android.example.di.modules

import com.minkiapps.shared.api.createApiService
import javax.inject.Singleton
import com.minkiapps.shared.api.ApiService
import dagger.Module
import dagger.Provides

@Module
class ApiModule {

    @Provides
    @Singleton
    fun providesApiService(): ApiService {
        return createApiService()
    }
}